bindings/node/lib/stateMachine.js

```diff
@@ -68,6 +68,11 @@ module.exports = function(modules) {
    * @class StateMachine
    */
   class StateMachine {
+    constructor(options) {
```

Comment:
why not just store `options` as `this.options` rather than making a sub document for `userBsonOptions`? I think there's value to a future where a flat options object is resolved from the connection string + MongoClient options, frozen, and then just passed all the way down the chain. Also this might eventually have options that aren't specifically bson options, you would have to have `userDecryptOptions`, `userEncryptOptions`, etc. I say it's turtles all the way down